lib/actions/field-trip.js

```diff
@@ -249,17 +406,355 @@ function planInbound (request) {
     }, config)
     // clearTrip()
     const queryParams = {
-      date: moment(request.travelDate).format(OTP_API_DATE_FORMAT),
+      date: getFormattedRequestTravelDate(request),
       departArrive: 'DEPART',
-      groupSize: getGroupSize(request),
       time: moment(request.leaveDestinationTime).format(OTP_API_TIME_FORMAT),
       ...locations
     }
     dispatch(setQueryParam(queryParams))
-    dispatch(routingQuery())
+    dispatch(makeFieldTripPlanRequests(request))
+  }
+}
+
+/**
+ * Makes appropriate OTP requests until enough itineraries have been found to
+ * accommodate the field trip group.
+ */
+function makeFieldTripPlanRequests (request) {
+  return async function (dispatch, getState) {
+    const prePlanState = getState()
+    const fieldTripModuleConfig = getModuleConfig(
+      prePlanState,
+      Modules.FIELD_TRIP
+    )
+
+    // request other known trip IDs that other field trips are using on the
+    // field trip request date
+    try {
+      await getTripIdsForTravelDate(dispatch, prePlanState, request)
```

Comment:
Since we're awaiting this statement, is it really necessary to pass dispatch and prePlanState in this way? I would prefer to make getTripIdsForTravelDate a standard thunk action.

Reply:
Refactored in a4480936984a2a1cb2d9dd6f97adc4ecd3a9a12c